compiler.rb

```diff
@@ -144,6 +145,7 @@
   end
 
   provider.generate output_path, types_to_generate, version
-  provider.copy_common_files output_path, version
 end
+
+provider.copy_common_files(output_path, version) unless provider.nil?
```

Comment:
So, if I understand this right - we fall through with our last-used product and generate the common files using that? I know this context will live in `git blame`, but I'd really like a comment here as well.

Reply:
Yeah that's about right. I'll add a comment in code.